keylocker.go

```diff
@@ -0,0 +1,76 @@
+package atoms
+
+import "sync"
+
+// KeyLocker allows to hold locks by key.
+type KeyLocker struct {
+	m   map[string]*sync.RWMutex
+	mux sync.RWMutex
+}
+
+// RWMutex returns a RWMutex for the specifc key.
+func (kl *KeyLocker) RWMutex(key string) (m *sync.RWMutex) {
+	kl.mux.RLock()
```

Comment:
Can you move this into a kl.get(key) func? That way you can re-use it for delete and RwMutex

Reply:
Can't reuse it for delete, the code is different :)